libraries/outputs.rb

```diff
@@ -24,6 +24,35 @@ def self.configure_outputs(node) # rubocop:disable Metrics/PerceivedComplexity,
         else
           [CernerSplunk.my_cluster(node)]
         end.each do |(cluster, bag)|
+          if bag['indexer_discovery'] == true
+            Chef::Log.warn "Configured ['receivers'] in cluster #{cluster} will be ignored since ['indexer_discovery'] is set to true." if bag['receivers']
+
+            indexer_discovery_settings = ((bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['outputs_configs']) || {}).reject do |k, _|
+              k.start_with?('_cerner_splunk')
+            end
+            output_stanzas["indexer_discovery:#{cluster}"] = indexer_discovery_settings
+
+            if bag['master_uri'].nil? || bag['master_uri'].empty?
+              Chef::Log.warn "master_uri is missing in the cluster databag: #{cluster}"
+            else
+              output_stanzas["indexer_discovery:#{cluster}"]['master_uri'] = bag['master_uri']
+            end
+            encrypt_password = CernerSplunk::ConfTemplate::Transform.splunk_encrypt node: node
+
+            pass =
+              if bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['pass4SymmKey']
+                bag['indexer_discovery_settings']['pass4SymmKey']
```

Comment:
is it me, or is this a symmetric key guaranteed to be plaintext ?

Reply:
> is this a symmetric key guaranteed to be plaintext on the chef server

If I understand the question, you are asking if we are going to have the pass4SymmKey as plain text in our data bags on the Chef server. I think that is how we have been dealing with other pass4Symmkeys (that we set in server.conf). Also based on the proposal made on an internal JIRA issue (which is in progress) investigating how to manage pass4Symmkey, the plan is to not encrypt pass4Symmkey on the forwarder because if we need to encrypt, we will either need to share the shared encryption key with all the forwarders or we will need to control the splunk.secret on the forwarders.